src/common/TrackedOp.h

```diff
@@ -162,15 +162,35 @@ class TrackedOp {
   uint32_t warn_interval_multiplier; // limits output of a given op warning
   // Transitions from false -> true without locks being held
   atomic<bool> is_tracked; //whether in tracker and out of constructor
+public:
+  Mutex comp_lock;
+  bool can_op_lock_comp;
+  void * rep_gather;
+  uint64_t tid; // for debugging
+  bool all_applied;
+  bool all_committed;
+  bool is_sent_ack;
+  bool completed;
```

Comment:
This is very much the wrong place for this.  The right answer is to leave this information in RepGather and arrange for that state to be separately locked.